### ISSUE TITLE
Wrap all usage of int built-in to prevent accidentally flooring due to floating point.

### ISSUE THREAD

**Issue by ehebert** [quantopian/zipline#281]
Related to https://github.com/quantopian/zipline/commit/a7c75403e8c026cfa8859ea68ee58a925af7d54c

So that a value like 30.0, when represented internally with a mantissa like 29.9999999998 does not get floored to 29 when passed to the `int()` function, wrap all usage of int using the logic in `slippage.almost_equal_to`.

This piece of work should probably also move `almost_equal_to` into `math_utils`, possibly creating a function like `to_int`.

**Comment by llllllllll** [comment]
@ehebert is this still an open issue?

**Comment by ehebert** [comment]
Yes, I think we would still benefit from aggressively protecting this edge case.
